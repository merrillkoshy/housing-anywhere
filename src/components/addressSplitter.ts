const addressSplitter = (address:string) => { 
    type Searchable = {
        [key: string]: string;
    };
    
        const chunk = address
            .replace("(", "")
            .replace(")", "")
            .replace('"', "")
            .split(",");
    
        const proto: Searchable = {
            housenumber: "",
            street: "",
            city: "",
            state: "",
            postalCode: "",
            neighborhood: "",
            countryCode: "",
        };
        chunk.forEach((element, i) => {
            var currentKey = Object.keys(proto)[i];
            proto[currentKey] = element
                .replace('"', "")
                .trim()
                .replace('"', "");
        } );
    return proto;
}
export default addressSplitter;