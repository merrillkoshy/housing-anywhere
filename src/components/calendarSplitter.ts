const calendarSplitter = (cal: string) => {
	type Searchable = {
		[key: string]: string | number | Array<Searchable>;
	};

	const regex = new RegExp(/\(([^()]*)\)/g);
	const isAnArray = regex.test(cal);
	const parentArrayPusher = [];
	const childArrayPusher: Array<any> = [];

	const chunk = cal
		.replace("(", "")
		.replace(")", "")
		.replace('"', "")
		.split(",");

	const proto: Searchable = {
		type: 0,
		dateFrom: "",
		dateTo: "",
	};
	chunk.forEach((element, i) => {
		var currentKey = Object.keys(proto)[i];
		const isAnArray = regex.test(element);
		proto[currentKey] = element
			.replace('"', "")
			.trim()
			.replace('"', "")
			.replace("{", "")
			.replace("}", "");

		if (isAnArray) {
			childArrayPusher.push(proto[currentKey]);
			proto[currentKey] = childArrayPusher;
		}
	});
	if (isAnArray) {
		parentArrayPusher.push(proto);
		return parentArrayPusher;
	} else {
		return proto;
	}
};
export default calendarSplitter;
