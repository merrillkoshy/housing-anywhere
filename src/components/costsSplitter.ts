const costsSplitter = (cost: string) => {
	type Searchable = {
		[key: string]: string | number | boolean | Array<Searchable>;
	};

	const regex = new RegExp(/\(([^()]*)\)/g);
	const isAnArray = regex.test(cost);
	const parentArrayPusher = [];
	const childArrayPusher: Array<any> = [];
	const chunk = cost
		.replace("(", "")
		.trim()
		.replace(")", "")
		.trim()
		.replace('"', "")
		.trim()
		.replace('\\"', "")
		.replaceAll(/\s/g, "")
		.split(",");

	const proto: Searchable = {
		type: "",
		value: 0,
		payableAt: "",
		payableBy: "",
		refundable: "",
		estimated: "",
		mandatory: "",
	};

	chunk.forEach((element, i) => {
		var currentKey = Object.keys(proto)[i];
		const isAnArray = regex.test(element);

		proto[currentKey] = element
			.replace("(", "")
			.trim()
			.replace(")", "")
			.replace("{", "")
			.replace("}", "")
			.replace('"', "")
			.replace(" ", "")
			.trim()
			.replace('"', "")
			.replaceAll(/\s/g, "");
		if (proto[currentKey] === "t") proto[currentKey] = true;
		if (proto[currentKey] === "f") proto[currentKey] = false;
		if (isAnArray) {
			childArrayPusher.push(proto[currentKey]);
			proto[currentKey] = childArrayPusher;
		}
	});

	if (isAnArray) {
		parentArrayPusher.push(proto);
		return parentArrayPusher;
	} else {
		return proto;
	}
};
export default costsSplitter;
