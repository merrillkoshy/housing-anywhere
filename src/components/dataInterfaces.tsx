import React from "react";

interface Address {
	housenumber: string;
	street: string;
	city: string;
	state: string;
	postalCode: string;
	neighborhood: string;
	countryCode: string;
}
interface Rules {
	pets: string;
	cleaningCommonRoomsIncluded: boolean;
	cleaningPrivateRoomIncluded: boolean;
	playMusic: string;
	smoking: string;
	couple: boolean;
}

interface AdditionalCosts {
	water: boolean;
	gas: boolean;
	electricity: boolean;
	internet: boolean;
}
interface Facilities {
	allergyFriendly: boolean;
	housematesGender: string;
	registrationPossible: boolean;
	tenantStatus: string;
	freePlaces: number;
	bedrooms: number;
	bedroomFurnished: boolean;
	bedroomSize: number;
	bedroomLock: boolean;
	balconyTerrace: string;
	basement: string;
	bathroom: string;
	garden: string;
	kitchen: string;
	livingRoom: string;
	parking: string;
	toilet: string;
	totalSize: number;
	wheelchairAccessible: boolean;
	airConditioning: boolean;
	bed: boolean;
	closet: boolean;
	desk: boolean;
	dishwasher: boolean;
	dryer: boolean;
	internet: boolean;
	roomFurniture: boolean;
	tv: boolean;
	washingMachine: boolean;
	wifi: boolean;
	flooring: string;
	heating: string;
	kitchenware: string;
	additionalCosts: AdditionalCosts;
}

interface Costs {
	type: string;
	value: string;
	payableAt: string;
	payableBy: string;
	refundable: boolean;
	estimated: boolean;
	mandatory: boolean;
}
interface Images {
	url: string;
}
interface CalendarOperations {
	type: number;
	dateFrom: string;
	dateTo: string;
}
export interface HAData {
	price: number;
	currencyCode: string;
	address: Address;
	type: number;
	kind: number;
	description: string;
	deposit: number;
	estimatedBills: number;
	minimumStayMonths: string;
	maxBookableDays: number;
	moveInWindow: number;
	currentOccupancy: number;
	rules: Rules;
	minAge: number;
	maxAge: number;
	preferredGender: string;
	alias: string;
	externalReference: string;
	extraData: string;
	facilities: Facilities;
	calendarOperations: CalendarOperations[];
	images: Images[];
	costs: Costs[];
}
