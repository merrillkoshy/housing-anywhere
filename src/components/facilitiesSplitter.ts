const facilitiesSplitter = (facilities: string) => {
	const proto: Searchable = {
		allergyFriendly: false,
		housematesGender: "",
		registrationPossible: false,
		tenantStatus: "",
		freePlaces: 0,
		bedrooms: 0,
		bedroomFurnished: false,
		bedroomSize: 0,
		bedroomLock: false,
		balconyTerrace: "",
		basement: "",
		bathroom: "",
		garden: "",
		kitchen: "",
		livingRoom: "",
		parking: "",
		toilet: "",
		totalSize: 0,
		wheelchairAccessible: false,
		airConditioning: false,
		bed: false,
		closet: false,
		desk: false,
		dishwasher: false,
		dryer: false,
		internet: false,
		roomFurniture: false,
		tv: false,
		washingMachine: false,
		wifi: false,
		flooring: "",
		heating: "",
		kitchenware: "",
	};
	const child: Searchable = {
		water: false,
		gas: false,
		electricity: false,
		internet: false,
	};

	if (facilities === null || facilities === "") return { proto, child };
	type Searchable = {
		[key: string]: string | boolean | number;
	};

	const chunk = facilities
		.replace("(", "")
		.trim()
		.replace(")", "")
		.trim()
		.replace('"', "")
		.trim()
		.replace('\\"', "")
		.replaceAll(/\s/g, "")
		.split(",");
	const additionalCosts = facilities
		.split('"(')[1]
		.replace(')"', "")
		.replace(")", "")
		.split(",");

	chunk.forEach((element, i) => {
		var currentKey = Object.keys(proto)[i];
		proto[currentKey] = element
			.replace("(", "")
			.trim()
			.replace(")", "")
			.replace('"', "")
			.replace(" ", "")
			.trim()
			.replace('"', "")
			.replaceAll(/\s/g, "");
		if (proto[currentKey] === "t") proto[currentKey] = true;
		if (proto[currentKey] === "f") proto[currentKey] = false;
	});
	additionalCosts.forEach((element, i) => {
		var currentKey = Object.keys(proto)[i];
		child[currentKey] = element.trim();
		if (child[currentKey] === "t") proto[currentKey] = true;
		if (child[currentKey] === "f") proto[currentKey] = false;
	});
	return { proto, child };
};
export default facilitiesSplitter;
