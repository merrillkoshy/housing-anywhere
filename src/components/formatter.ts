const formatter = (element: any): string => {
	var temp = "";
	Object.values(element).map((val: any, i: number) => {
		if (i !== Object.keys(element).length - 1) {
			if (val === "") return (temp = temp + "null,");
			return (temp = temp + val.toString() + ",");
		}
		if (typeof val === "object")
			return (temp = temp + '"' + formatter(val) + '"');
		if (val === "") return (temp = temp + "null");
		return (temp = temp + val);
	});
	return "(" + temp + ")";
};
const arrayformatter = (element: any): any[] => {
	var temp = "";
	var innertemp = [];
	innertemp.push(
		element.map((el: any) => {
			Object.values(el).map((val: any, i: number) => {
				if (i !== Object.keys(el).length - 1)
					return (temp = temp + val.toString() + ",");
				return (temp = temp + val);
			});
			return "(" + temp + ")";
		})
	);

	return innertemp.flat();
};
const formatters = { formatter, arrayformatter };
export default formatters;
