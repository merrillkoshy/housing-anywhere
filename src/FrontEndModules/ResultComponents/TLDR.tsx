import React from "react";
import { Costs, Rules } from "../../components/dataInterfaces";
import Costing from "./Costing";
import RulesComponent from "./Rules";

const TLDR = ({
	description,
	currencycode,
	price,
	costs,
	rules,
}: {
	description: string;
	currencycode: string;
	price: number;
	costs: Costs[];
	rules: Rules;
}) => {
	return (
		<>
			<section id="description">
				<h3>Description</h3>
				<p id="description">{description}</p>
			</section>
			<section className="d-flex" id="cost">
				<>
					<div className="col-1 pricing-text">
						<strong id="currencycode">{currencycode}</strong>
					</div>
					<div className="col mx-2 pricing-text">
						<strong id="price">
							{price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ".")}
						</strong>
					</div>
					<div className="col-8 mx-2 ">
						<Costing costsArray={costs} />
					</div>
				</>
			</section>
			<section className="d-flex" id="rules">
				<RulesComponent rules={rules} />
			</section>
		</>
	);
};
export default TLDR;
