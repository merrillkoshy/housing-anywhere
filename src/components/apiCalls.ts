import axios from "axios";

import { HAData } from "./dataInterfaces";

const insertData = (data: HAData) => {
	console.log(data);
	Object.keys(data).map((obs, i) => {
		console.log(obs, i);
	});
	console.log(process.env.REACT_APP_DB_URL);
	axios
		.post(`${process.env.REACT_APP_DB_URL}/listing`, data, {
			headers: {
				jwtSecret: "merrillkoshy",
			},
		})
		.then((res) => {
			console.log(res);
		});
};
export default insertData;
