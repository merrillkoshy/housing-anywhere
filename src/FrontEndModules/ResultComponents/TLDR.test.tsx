import React, { SetStateAction } from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom/extend-expect";
import TLDR from "./TLDR";
import { Rules } from "../../components/dataInterfaces";

describe("Description module", () => {
	const rules: Rules = {
		pets: "",
		cleaningCommonRoomsIncluded: false,
		cleaningPrivateRoomIncluded: false,
		playMusic: "",
		smoking: "",
		couple: false,
	};
	const TLDRModule = shallow(
		<TLDR
			description={""}
			currencycode={""}
			price={0}
			costs={[]}
			rules={rules}
		/>
	);
	it("renders", () => {
		expect(TLDRModule).toMatchSnapshot();
	});

	it("finds a description paragraph", () => {
		expect(TLDRModule.find("#description").exists()).toBe(true);
	});
	it("finds a currencycode div", () => {
		expect(TLDRModule.find("#currencycode").exists()).toBe(true);
	});
	it("finds a price div", () => {
		expect(TLDRModule.find("#price").exists()).toBe(true);
	});
});
