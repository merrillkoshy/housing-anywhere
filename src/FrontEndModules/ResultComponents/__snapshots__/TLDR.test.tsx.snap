// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description module renders 1`] = `
<Fragment>
  <section
    id="description"
  >
    <h3>
      Description
    </h3>
    <p
      id="description"
    />
  </section>
  <section
    className="d-flex"
    id="cost"
  >
    <div
      className="col-1 pricing-text"
    >
      <strong
        id="currencycode"
      />
    </div>
    <div
      className="col mx-2 pricing-text"
    >
      <strong
        id="price"
      >
        0
      </strong>
    </div>
    <div
      className="col-8 mx-2 "
    >
      <Costing
        costsArray={Array []}
      />
    </div>
  </section>
  <section
    className="d-flex"
    id="rules"
  >
    <RulesComponent
      rules={
        Object {
          "cleaningCommonRoomsIncluded": false,
          "cleaningPrivateRoomIncluded": false,
          "couple": false,
          "pets": "",
          "playMusic": "",
          "smoking": "",
        }
      }
    />
  </section>
</Fragment>
`;
